### PR TITLE
Add one-argument division and rationals

### PR DESCRIPTION
**Pull request by kirbyfan64**
Ref. #825 and #826.

### CODE REVIEW

**Comment by Foxboron** [comment]
Missing tests and docs \o/
Does looks OK apart from that.

**Comment by kirbyfan64** [comment]
@Foxboron I knew I forgot something!

**Comment by kirbyfan64** [comment]
Now everything's good!

File: hy/lex/parser.py [hunk 1]
```diff
@@ -252,6 +252,14 @@ def t_identifier(p):
     except ValueError:
         pass
 
+    if '/' in obj:
+        try:
+            lhs, rhs = obj.split('/')
+            return HyExpression([HySymbol('fraction'), HyInteger(lhs),
+                                 HyInteger(rhs)])
+        except:
```
Comment (zackmdavis) [inline hy/lex/parser.py]: `except ValueError:`

(In accordance with the platitude that it's usually better to catch only the specific exceptions the programmer actually expects to happen, giving us the opportunity to notice and correct the erroneous program if something else happens,  and a bare `except:` will swallow even `SystemExit` and `KeyboardInterrupt`. Here, we get a `ValueError` both when `obj` has a non-one number of slashes, and when `lhs` or `lrs` can't be interpreted as an integer; I can't think of any other exceptions to expect.)

File: hy/compiler.py [hunk 2]
```diff
@@ -1718,11 +1716,13 @@ def compile_maths_expression(self, expression):
...
     @builds("+")
     @builds("*")
+    @builds("/")
+    @builds("//")
     def compile_maths_expression_mul(self, expression):
         if len(expression) > 2:
             return self.compile_maths_expression(expression)
         else:
-            id_op = {"+": HyInteger(0), "*": HyInteger(1)}
+            id_op = {"+": HyInteger(0), "*": HyInteger(1), "/": HyInteger(1)}
```
Comment (zackmdavis) [inline hy/compiler.py]: Floor division needs its identity, too&mdash;
```
=> (// 2)
Traceback (most recent call last):
  File "/home/zmd/Code/hy/hy/hy/compiler.py", line 2315, in hy_compile
    result = compiler.compile(tree)
  File "/home/zmd/Code/hy/hy/hy/compiler.py", line 428, in compile
    ret = self.compile_atom(_type, tree)
  File "/home/zmd/Code/hy/hy/hy/compiler.py", line 420, in compile_atom
    ret = _compile_table[atom_type](self, atom)
  File "/home/zmd/Code/hy/hy/hy/compiler.py", line 596, in compile_raw_list
    ret = self._compile_branch(entries)
  File "/home/zmd/Code/hy/hy/hy/compiler.py", line 483, in _compile_branch
    return _branch(self.compile(expr) for expr in exprs)
  File "/home/zmd/Code/hy/hy/hy/compiler.py", line 313, in _branch
    results = list(results)
  File "/home/zmd/Code/hy/hy/hy/compiler.py", line 483, in <genexpr>
    return _branch(self.compile(expr) for expr in exprs)
  File "/home/zmd/Code/hy/hy/hy/compiler.py", line 440, in compile
    raise_empty(HyCompileError, e, sys.exc_info()[2])
  File "<string>", line 1, in raise_empty
hy.errors.HyCompileError: Internal Compiler Bug 😱
⤷ KeyError: '//'
Compilation traceback:
File "/home/zmd/Code/hy/hy/hy/compiler.py", line 428, in compile
    ret = self.compile_atom(_type, tree)
  File "/home/zmd/Code/hy/hy/hy/compiler.py", line 420, in compile_atom
    ret = _compile_table[atom_type](self, atom)
  File "/home/zmd/Code/hy/hy/hy/compiler.py", line 1817, in compile_expression
    ret = self.compile_atom(fn, expression)
  File "/home/zmd/Code/hy/hy/hy/compiler.py", line 420, in compile_atom
    ret = _compile_table[atom_type](self, atom)
  File "/home/zmd/Code/hy/hy/hy/compiler.py", line 1733, in compile_maths_expression_mul
    ]).replace(expression)
```

Comment (kirbyfan64) [inline hy/compiler.py]: @zackmdavis Fixed.

File: hy/lex/parser.py (hunk 1 above)
Comment (paultag) [inline hy/lex/parser.py]: Yeah, let's fix that `except`

**Comment by paultag** [comment]
Nice work so far. This is going to be great!

File: hy/lex/parser.py (hunk 1 above)
Comment (kirbyfan64) [inline hy/lex/parser.py]: @paultag @zackmdavis Fixed.

**Comment by kirbyfan64** [comment]
Was there anything else that needed to be addressed? I'd try resolving the merge conflicts, but, considering how the last time I tried that went, I doubt that would be a good idea...

**Comment by gilch** [comment]
I'm wondering if we should auto-import anything else from the `fractions` module to make the fractions work properly.

**Comment by zackmdavis** [comment]
I'm ***+1*** on this. 

@kirbyfan64 [This is what I get](https://github.com/hylang/hy/compare/master...zackmdavis:div_rebased) when I rebase on master, resolve the conflict in docs/tutorial.rst, and squash "Fix Flake8", "Add floor division identity", and "Remove bare except clause" onto the first commit. OK with you?

**Comment by kirbyfan64** [comment]
@zackmdavis Well, if you're already going through the pain^H^H^H^Hawesomeness of `git rebase`, you might as well squash them all into one commit. ;)

Of course, I can just do it myself later, too.

**Comment by kirbyfan64** [comment]
Squashed and rebased!

**Comment by Foxboron** [comment]
I feel `1/2` should be mentioned in the docs. Minor nitpick maybe? It looks great apart from that tho!

**Comment by kirbyfan64** [comment]
@Foxboron Where?

**Comment by kirbyfan64** [comment]
@Foxboron Fixed.

**Comment by Foxboron** [comment]
Yes. +1+1+1+1 all over this!

:shipit:

**Pull request merged**